api/v1beta1/common_types.go

```diff
@@ -122,6 +122,14 @@ const (
 	ReplicasManagedByAnnotation = "cluster.x-k8s.io/replicas-managed-by"
 )
 
+// NodeUninitializedTaint is added to the nodes at creation when using KubeadmBootstrap provider.
+// This taint is added to prevent workloads to be scheduled on the nodes till the labels are initially
+// synced form the machines to the nodes. The taint is removed form the node after the node labels are synced.
```

Comment:
```suggestion
// NodeUninitializedTaint can be added to Nodes at creation by the bootstrap provider, e.g. the
// KubeadmBootstrap provider will add the taint.
// This taint is used to prevent workloads to be scheduled on Nodes before the node is initialized by Cluster API.
// As of today the Node initializiation consists of syncing labels from Machines to Nodes. Once the labels
// have been initially synced the taint is removed form the Node.
```

Just to write it a bit more generic for other bootstrap providers + expand the scope to "node initializiation" in general, of which label sync is just one part (even if today it's the only thing we do)